File: front/src/app/models/croisement.ts
import { Creneau } from "./creneau";
import { Stand } from "./stand";
import { Benevole } from "./benevole";
import { Evenement } from './evenement';

export class Croisement {
  id:number;
  creneau: Creneau;
  stand: Stand;
  benevoles: Benevole[];
  selected:boolean;
  besoin:boolean;
  limite:number;
  evenement:Evenement;


}

File: app/src/main/java/com/wild/corp/model/Croisement.java
package com.wild.corp.model;



import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;


import java.util.List;

@Entity
@Data
@NoArgsConstructor
@Table(name = "CROISEMENT")
@JsonIgnoreProperties(ignoreUnknown = true)
public class Croisement {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID", nullable = false)
    private Integer id;

    @Column(name = "LIMITE", nullable = false)
    private Integer limite;

    @Column(name = "BESOIN", nullable = false)
    private Boolean besoin;

    @ManyToMany(mappedBy="croisements")
    @JsonIgnoreProperties({"croisements", "evenement", "telephone", "email"})
    private List<Benevole> benevoles;

    @ManyToOne
    @JsonIgnoreProperties({"croisements"})
    private Stand stand;

    @ManyToOne
    @JsonIgnoreProperties({"croisements"})
    private Creneau creneau;



}
